### PR TITLE
Fix/optimization

### PR DESCRIPTION
**Pull request by dhruv1255**
Optimized the code by adding atmoic transactions

### CODE REVIEW

File: portal/views.py [hunk 1]
```diff
@@ -73,13 +73,13 @@ def test_page(request):
...
 
-    for i in range(1, len(question)+1):
-        nums.append({
-            'question_id': questions[i-1]["id"],
-            'i': i,
-        })
+    # for i in range(1, len(question)+1):
+    #     nums.append({
+    #         'question_id': questions[i-1]["id"],
+    #         'i': i,
```
Comment (dhruv1255) [inline portal/views.py]: remove comments

File: portal/views.py [hunk 2]
```diff
@@ -73,13 +73,13 @@ def test_page(request):
...
-    nums = []
+    # nums = []
 
-    for i in range(1, len(question)+1):
-        nums.append({
-            'question_id': questions[i-1]["id"],
-            'i': i,
-        })
+    # for i in range(1, len(question)+1):
+    #     nums.append({
```
Comment (samarth4445) [inline portal/views.py]: remove comments

File: portal/views.py [hunk 3]
```diff
@@ -141,6 +136,58 @@ def user_page(request):
...
+    time_d = datetime(year=1, month=1, day=1, hour=time_c.hour,
+                      minute=time_c.minute, second=time_c.second)
+    time_diff2 = time_b-time_d
+
+    test_hour = TestHour.objects.get(test=test)
+
+    time_diff = timedelta(hours=test_hour.time.hour, minutes=test_hour.time.minute,
+                          seconds=test_hour.time.second) - time_diff2
+
+    print(time_diff.seconds)
```
Comment (samarth4445) [inline portal/views.py]: remove prints

File: portal/views.py [hunk 4]
```diff
@@ -407,16 +396,13 @@ def detailed_result(request, userID, testID):
...
 
     user_answers_list : UserAnswers = []
 
     for user_answer in user_answers:
         if user_answer.question in questions:
             user_answers_list.append(user_answer)
-    
-    print(user_answers_list)
-    
+
```
Comment (samarth4445) [inline portal/views.py]: remove unnecessary lines

File: portal/views.py [hunk 5]
```diff
@@ -169,12 +217,62 @@ def admin_panel(request):
...
+    if (not request.user.is_superuser):
+        return HttpResponseForbidden('You are not allowed to access this resource!')
+
+    test = Test.objects.filter(id=testID).first()
+
+    return render(request, 'portal/test-settings/time-settings.html', {
+        'test': test
     })
 
+def signup_user(request):
```
Comment (samarth4445) [inline portal/views.py:283]: why have you not added the required html file in the render here for signup? Please render the requested html file.

Comment (dhruv1255) [reply portal/views.py:283]: The signup HTML file has not been added by the frontend team yet, I will add it as soon as the file is available.

Comment (samarth4445) [reply portal/views.py:283]: Ok, then for now we will  merge this branch.

**Pull request merged**